docs/azure/sdk/snippets/dependency-injection/HostApplicationBuilder/Program.cs

```diff
@@ -1,15 +1,50 @@
 ﻿using Azure.Identity;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
 using Microsoft.Extensions.Azure;
 using Microsoft.Extensions.Hosting;
 
-HostApplicationBuilder builder = Host.CreateApplicationBuilder(args);
+List<string> queueNames = await GetQueueNames();
 
-builder.Services.AddAzureClients(clientBuilder =>
-{
-    clientBuilder.AddSecretClient(new Uri("<key_vault_url>"));
-    clientBuilder.AddBlobServiceClient(new Uri("<storage_url>"));
-    clientBuilder.UseCredential(new DefaultAzureCredential());
-});
+IHost host = Host.CreateDefaultBuilder(args)
+    .ConfigureServices(services =>
+    {
+        services.AddAzureClients(clientBuilder =>
+        {
+            // Register clients for each service
+            clientBuilder.AddSecretClient(new Uri("<key_vault_url>"));
+            clientBuilder.AddBlobServiceClient(new Uri("<storage_url>"));
+            clientBuilder.AddServiceBusClientWithNamespace("<your_namespace>.servicebus.windows.net");
+            clientBuilder.UseCredential(new DefaultAzureCredential());
+
+            // Register subclients for Service Bus
+            foreach (string queueName in queueNames)
+            {
+                clientBuilder.AddClient<ServiceBusSender, ServiceBusClientOptions>((_, _, provider) =>
+                provider.GetService(typeof(ServiceBusClient)) switch
+                {
+                    ServiceBusClient client => client.CreateSender(queueName),
+                    _ => throw new InvalidOperationException("Unable to create ServiceBusClient")
+                }).WithName(queueName);
+            }
+        });
+    }).Build();
 
-using IHost host = builder.Build();
 await host.RunAsync();
+
+async Task<List<string>> GetQueueNames()
+{
+    // Query the available queues for the Service Bus namespace.
+    var adminClient = new ServiceBusAdministrationClient
+        ("<your_namespace>.servicebus.windows.net", new DefaultAzureCredential());
+    var queueNames = new List<string>();
+
+    // Because the result is async, the queue names need to be captured to a standard list to avoid async
+    // calls when registering. Failure to do so results in an error with the services collection.
```

Comment:
```suggestion
    // Because the result is async, the queue names need to be captured
    // to a standard list to avoid async calls when registering. Failure to 
    // do so results in an error with the services collection.
```